src/OpenTelemetry/Logs/OpenTelemetryLogger.cs

```diff
@@ -51,29 +53,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             var processor = provider.Processor;
             if (processor != null)
             {
+                var activity = Activity.Current;
+
                 var pool = provider.LogRecordPool;
 
                 var record = pool.Rent();
 
-                record.ScopeProvider = provider.IncludeScopes ? this.ScopeProvider : null;
-                record.State = provider.ParseStateValues ? null : state;
-                record.StateValues = provider.ParseStateValues ? ParseState(record, state) : null;
-                record.BufferedScopes = null;
+                ref LogRecord.LogRecordILoggerData iloggerData = ref record.ILoggerData;
+
+                iloggerData.TraceState = provider.IncludeTraceState && activity != null
+                    ? activity.TraceStateString
+                    : null;
+                iloggerData.CategoryName = this.categoryName;
+                iloggerData.EventId = eventId;
+                iloggerData.LogLevel = logLevel;
+                iloggerData.Exception = exception;
+                iloggerData.ScopeProvider = provider.IncludeScopes ? this.ScopeProvider : null;
+                iloggerData.BufferedScopes = null;
 
                 ref LogRecordData data = ref record.Data;
 
                 data.TimestampBacking = DateTime.UtcNow;
-                data.CategoryName = this.categoryName;
-                data.LogLevel = logLevel;
-                data.EventId = eventId;
-                data.Message = provider.IncludeFormattedMessage ? formatter?.Invoke(state, exception) : null;
-                data.Exception = exception;
 
-                LogRecordData.SetActivityContext(ref data, Activity.Current);
+                LogRecordData.SetActivityContext(ref data, activity);
+
+                var attributes = record.Attributes = provider.IncludeState
+                    ? ProcessState(record, ref iloggerData, in state, provider.ParseStateValues)
+                    : null;
+
+                if (!TryGetOriginalFormatFromAttributes(attributes, out var originalFormat))
+                {
+                    var formattedMessage = formatter?.Invoke(state, exception) ?? state?.ToString();
+
+                    data.Body = formattedMessage;
+                    iloggerData.FormattedMessage = formattedMessage;
```

Comment:
Why change this behavior of setting `FormattedMessage` even when `IncludeFormattedMessage` is set to false?

Reply:
See issue description. Basically, what good is a log to anyone which has no message or template? 😄

Reply:
> Basically, what good is a log to anyone which has no message or template? 

Shouldn't that be covered by `LogRecord.Body`? Why not continue to respect `IncludeFormattedMessage` to decide if `LogRecord.FormattedMessage` should be populated? (I did go through the description. 😀)

Reply:
Gotcha. Sorry wasn't following your question.

If we call the formatter or state.ToString to generate a formatted message either because a) Body/{OriginalFormat} was not found or b) `IncludeFormattedMessage == true` than the result is set to `LogRecord.FormattedMessage`. [Documented in the XML comments](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334/files#diff-f2353033f9d1a52c7a15487b3753305816327a63b88130c2a656b59297d91223R174-R182). Also on the XML for [IncludeFormattedMessage](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334/files#diff-34b4c998e6fd43814d6c6b494ff5dfc3d009cee4e83a97574656e3efe6ac1b3bR40-R51).

Why do that? Well the primary reason I suppose is nothing understands `LogRecord.Body` today. Presumably things understand `LogRecord.FormattedMessage` correctly 🤣 Secondary reason is it just felt like the correct behavior to me. We formatted a message so it goes on the spot for that. Maybe I'm overthinking it.

Reply:
I think adding this caveat complicates the meaning of the option. Going by the name `IncludeFormattedMessage`, intuitively it would feel like a simple yes/no with:

yes => FormattedMessage is populated
no => FormattedMessage is not populated

Also, isn't this technically a breaking change?

Again, if we want to discourage people from using `FormattedMessage` and start using `Body` then we should probably mark `FormattedMessage` obsolete and continuing to offer the existing functionality for it. It goes back to my comment here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334#discussion_r1160250113

Reply:
> if we want to discourage people from using FormattedMessage

My intention is definitely not to discourage anyone from using it! It is expensive so it is off by default, but if users are willing to pay for it, perfectly happy for them to do it 😄 `Body` and `FormattedMessage` are different things. They could end up the same for some edge cases but for most log messages `Body` is going to be the template and `FormattedMessage` is only there if it is requested. Intention was in no way for `FormattedMessage` to be obsoleted.

> Also, isn't this technically a breaking change?

Technically yes. But I view it more as a bug fix. Today we can export logs with no message at all. This fixes that. Risk seems low to me the break would cause any problems. Can you think of any situations where it might be a problem? OTLP seems OK. Users will just see a populated Body where it was null before.

Keep in mind most log messages will have {OriginalFormat} it is rare in ILogger to not have it. You have to be doing Log<T> calls yourself with custom states where you didn't spend the time to add it in.